api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -340,6 +340,18 @@ default boolean dropTable(TableIdentifier identifier) {
    */
   Table loadTable(TableIdentifier identifier);
 
+  /**
+   * Register a table with the catalog if it does not exist.
```

Comment:
FYI @bryanck. What do you think about this?

Reply:
LGTM, it is very straightforward, I just had one question below...